bin/check-marathon-apps.rb

```diff
@@ -0,0 +1,398 @@
+
+    apps.each do |app|
+      # Select app queue if any
+      app_queue = queue.select { |q| q['app']['id'][/^#{app['id']}$/] }.to_a.first
+
+      # Build check result
+      check_result = check_result_scaffold(app)
+
+      # Parse Marathon app labels
+      labels_config = parse_app_labels(app['labels'].to_h)
+
+      REFERENCES.each do |reference|
+        # / is and invalid character
+        check_result['name'] = "check_marathon_app#{app['id'].tr('/', '_')}_#{reference}"
+
+        state = case reference
+                when 'health'
+                  get_marathon_app_health(app)
+                when 'status'
+                  get_marathon_app_status(app, app_queue.to_h)
+                end
+
+        # Merge user provided check config
+        check_result.merge!(check_config.dig('_').to_h)
+        check_result.merge!(check_config.dig(reference, '_').to_h)
+        check_result.merge!(check_config.dig(reference, state).to_h)
+
+        # Merge Marathon parsed check config
+        check_result.merge!(labels_config.dig('_').to_h)
+        check_result.merge!(labels_config.dig(reference, '_').to_h)
+        check_result.merge!(labels_config.dig(reference, state).to_h)
+
+        # Build check result output
+        check_result['output'] = "#{reference.upcase} #{state.capitalize} - "\
+          "tasksRunning(#{app['tasksRunning'].to_i}), tasksStaged(#{app['tasksStaged'].to_i}), "\
+          "tasksHealthy(#{app['tasksHealthy'].to_i}), tasksUnhealthy(#{app['tasksUnhealthy'].to_i})"
+
+        # Make sure that check result data types are correct
+        sanitize_check_result(check_result)
+
+        # Send the result to sensu-client HTTP socket
+        post_check_result(check_result)
+      end
+    end
+
+    ok 'Marathon Apps Status and Health check is running properly'
+  end
+
+  def check_result_scaffold(app)
+    {
+      'name' => '',
+      'executed' => Time.now.to_i,
+      'marathon' => app.select { |k, _| config[:marathon_keys].split(',').include?(k) },
+      'source' => 'marathon',
+      'output' => '',
+      'status' => 3
+    }
+  end
+
+  def sanitize_check_result(check_result)
+    # Force data types of different fields on the check result
+    # https://sensuapp.org/docs/latest/reference/checks.html#example-check-definition
+    # https://sensuapp.org/docs/latest/reference/checks.html#check-result-specification
+    check_result.each do |k, v|
+      if %w(publish standalone auto_resolve force_resolve handle truncate_output).include?(k)
+        # Boolean
+        check_result[k] = v.to_s.eql?('true')
+      elsif %w(status interval issued executed timeout ttl ttl_status low_flap_threshold high_flap_threshold truncate_output_length).include?(k)
+        # Integer
+        check_result[k] = Integer(v)
+      elsif %w(subscribers handlers aggregates).include?(k)
+        # Array
+        check_result[k] = Array(v.split(','))
+      end
+    end
+  end
+
+  def get(path)
```

Comment:
thats a pretty generic name could we use something like `rest_client` or similar?

Reply:
makes sense, updating